internal/config/identity/ldap/ldap.go

```diff
@@ -56,6 +57,123 @@ func (l *Config) LookupUserDN(username string) (string, []string, error) {
 	return bindDN, groups, nil
 }
 
+// DoesUsernameExist checks if the given username exists in the LDAP directory.
+// The given username could be just the short "login" username or the full DN.
+// When the username is found, the full DN is returned, otherwise the returned
+// string is empty. If the user is not found, err = nil, otherwise, err != nil.
+func (l *Config) DoesUsernameExist(username string) (string, error) {
+	conn, err := l.LDAP.Connect()
+	if err != nil {
+		return "", err
+	}
+	defer conn.Close()
+
+	// Bind to the lookup user account
+	if err = l.LDAP.LookupBind(conn); err != nil {
+		return "", err
+	}
+
+	// Check if the passed in username is a DN existing in the LDAP tree.
+	if parsedUsernameDN, err := ldap.ParseDN(username); err == nil {
```

Comment:
Swap so you handle the `err != nil` case and have the happy (longer) path to the left.